R/read.R

```diff
@@ -242,7 +242,7 @@ st_read.character = function(dsn, layer, ..., query = NA, options = NULL, quiet
 proc_zip <- function(dsn){
 	# create unique directory so it can be deleted
 	tmp <- paste0(tempdir(),"/shp")
```

Comment:
I think that using `tempfile()` here would be safer than `paste0(tempdir(), "/shp")`, since the latter is constant for the whole R session, and may already exist and contain data. `exdir = ...` creates the directory if it doesn't already exist.

Reply:
Cool! Updated the file.